### PR TITLE
Abandon DarwiniaPrecompiles tuple

### PR DESCRIPTION
**Pull request by AsceticBear**
We can specify precompile contract address as any value after this pull request merged.

### CODE REVIEW

File: frame/evm/precompile/src/lib.rs [hunk 1]
```diff
@@ -18,28 +18,40 @@
...
+			a if a == to_address(22) => Some(Kton::<R>::execute(input, target_gas, context)),
+			a if a == to_address(23) => Some(Issuing::<R>::execute(input, target_gas, context)),
+			_ => None,
+		}
+	}
+}
+
+fn to_address(a: u64) -> H160 {
+	H160::from_low_u64_be(a)
+}
```
Comment (HackFisher) [inline frame/evm/precompile/src/lib.rs:57]: I like this approach, but I think this code should located in pangolin runtime.

**Review by HackFisher — CHANGES_REQUESTED** [review]

File: frame/evm/precompile/src/lib.rs (hunk 1 above)
Comment (AsceticBear) [reply frame/evm/precompile/src/lib.rs]: Great catch. It would be much better to put this logic in runtime. Please review again.

**Review by HackFisher — APPROVED** [review]

**Comment by HackFisher** [comment]
@AurevoirXavier @xiaoch05 please help review this.

**Review by xiaoch05 — APPROVED** [review]

File: bin/node/runtime/pangolin/src/pallets/evm.rs [hunk 2]
```diff
@@ -1,11 +1,46 @@
...
+		input: &[u8],
+		target_gas: Option<u64>,
+		context: &Context,
+	) -> Option<core::result::Result<(ExitSucceed, Vec<u8>, u64), ExitError>> {
+		fn to_address(a: u64) -> H160 {
+			H160::from_low_u64_be(a)
+		}
+		match address {
+			// Ethereum precompiles
+			a if a == to_address(1) => Some(ECRecover::execute(input, target_gas, context)),
```
Comment (AurevoirXavier) [inline bin/node/runtime/pangolin/src/pallets/evm.rs:30]: `_ if address == to_address(1)` looks better

File: bin/node/runtime/pangolin/Cargo.toml [hunk 3]
```diff
@@ -13,6 +13,7 @@ version    = "2.4.0"
 # crates
 array-bytes       = { version = "1.1.0" }
 codec             = { package = "parity-scale-codec", version = "2.1.1", default-features = false }
+evm-crate         = { default-features = false, version = "0.25.0", features = ["with-codec"], package = "evm" }
```
Comment (AurevoirXavier) [inline bin/node/runtime/pangolin/Cargo.toml:16]: Any reason rename evm to evm-crate?

Comment (AsceticBear) [reply bin/node/runtime/pangolin/Cargo.toml:16]: O, that's because the `crate evm` will conflict with the`runtime mod evm`.

**Review by AurevoirXavier — APPROVED** [review]

**Pull request merged**